embedded-graphics/src/primitives/rectangle.rs

```diff
@@ -107,6 +106,19 @@ impl Rectangle {
 
         self.top_left + Size::new(dx, dy)
     }
+
+    /// Returns the bottom right corner of this rectangle.
+    ///
+    /// Because the smallest rectangle that can be represented by its corners
+    /// has a size of 1 x 1 pixel this functions returns `None` if the width or
+    /// height of the rectangle is zero.
+    pub fn bottom_right(&self) -> Option<Point> {
```

Comment:
What's the reasoning behind returning an `Option` and not clamping the `Point` to `self.top_left`? If the rectangle has a size of zero, AIUI from #182 wouldn't `bottom_right == top_left`?

Reply:
This is mentioned in the "Problems" paragraph in #182. `bottom_right == top_left` is a rectangle of `Size::new(1, 1)`.

I would expect that this code works for all rectangles:
```rust
let r = Rectangle::new(...);
assert_eq!(r, Rectangle::with_corners(r.top_left, r.bottom_right.unwrap());
```

If `bottom_right` is clamped to `top_left` it would return the wrong value for zero sized rectangles. This means that the code above would convert a rectangle with `Size::zero()` to a rectangle with `Size::new(1, 1)`.

Reply:
I read throught that bullet but didn't quite get the meaning behind it, but I see what you mean now. Makes sense - thanks for explaining 👍